docs/DeveloperGuide.adoc

```diff
@@ -153,6 +158,33 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+// tag::undobyreverse[]
+=== [Implemented] Undo Feature
+==== Current Implementation Logic
+
+The undo function uses the revert command method without using states and history, unlike the proposed method.
+This is because an issue was encountered with referencing lists and firing reminders multiple times when the state history method was used.
+
+In this implementation, the commands that can be undone; that is, all the commands except `UndoCommand`, `ExitCommand`,
+`UpCommand` and `DownCommand` now extend from an abstract class `UndoableCommand`, which is a subclass of Command.
+Subclasses of `UndoableCommand` must implement a method `reverse(ItemModel model)`, which should do the exact opposite
+of the `execute(ItemModel model)` in that Command.
+
+The command execution history is stored in a stack, which is maintained in `ElisaCommandHistory`.
+
+Below is a possible usage scenario and the app behaviour.
+
+Step 1. The user executes `task eat`. A task with description "eat" is added and then the command is pushed into the commands stack.
+
+image::UndoStackStep1.png[]
```

Comment:
Maybe you could label `AddTaskCommand` in the model to reflect `task eat` command in step 1. Perhaps the image could be resized so that the text size is consistent with the DG.